CrewChiefV4/iRacing/iRacingGameStateMapper.cs

```diff
@@ -121,14 +121,8 @@ public GameStateData mapToGameStateData(Object memoryMappedFileStruct, GameState
                 prevTrackSurface = playerCar.Live.TrackSurface.ToString();
             }*/
 
-            Boolean sessionOfSameTypeRestarted = ((currentGameState.SessionData.SessionType == SessionType.Race && lastSessionType == SessionType.Race) ||
-                (currentGameState.SessionData.SessionType == SessionType.Practice && lastSessionType == SessionType.Practice) ||
-                (currentGameState.SessionData.SessionType == SessionType.Qualify && lastSessionType == SessionType.Qualify)) &&
-                ((lastSessionPhase == SessionPhase.Green || lastSessionPhase == SessionPhase.FullCourseYellow) || lastSessionPhase == SessionPhase.Finished) &&
-                (currentGameState.SessionData.SessionPhase == SessionPhase.Countdown);
-
-            if (sessionOfSameTypeRestarted || currentGameState.SessionData.SessionType != SessionType.Unavailable
-                && lastSessionPhase != SessionPhase.Countdown
+
+            if (currentGameState.SessionData.SessionType != SessionType.Unavailable && shared.Telemetry.IsNewSession
                 && lastSessionType != currentGameState.SessionData.SessionType || sessionNumber != previousSessionNumber || previousSessionId != currentGameState.SessionData.SessionId)
```

Comment:
might be worth putting brackets around these so it's
if (currentGameState.SessionData.SessionType != SessionType.Unavailable && shared.Telemetry.IsNewSession
                  && (lastSessionType != currentGameState.SessionData.SessionType || sessionNumber != previousSessionNumber || previousSessionId != currentGameState.SessionData.SessionId))

Unless I'm mis-reading the logic